packages/pluggableWidgets/bar-chart-native/src/components/BarChart.tsx

```diff
@@ -0,0 +1,252 @@
+import { Legend } from "./Legend";
+
+export interface BarChartProps {
+    series: BarChartSeries[];
+    sortOrder: SortOrderEnum;
+    style: BarChartStyle;
+    presentation: string;
+    showLegend: boolean;
+    showLabels: boolean;
+    xAxisLabel?: string;
+    yAxisLabel?: string;
+    warningPrefix?: string;
+}
+
+export interface BarChartSeries {
+    dataPoints: BarDataPoints;
+    xFormatter?: (xValue: number | Date | string) => string;
+    yFormatter?: (yValue: number | Date | string) => string;
+    name?: string;
+}
+
+export type BarDataPoints =
+    | Array<BarDataPoint<number, number>>
+    | Array<BarDataPoint<number, Date>>
+    | Array<BarDataPoint<Date, number>>
+    | Array<BarDataPoint<Date, Date>>
+    | Array<BarDataPoint<string, string>>
+    | Array<BarDataPoint<string, number>>
+    | Array<BarDataPoint<string, Date>>
+    | Array<BarDataPoint<number, string>>
+    | Array<BarDataPoint<Date, string>>;
+
+export interface BarDataPoint<X extends number | Date | string, Y extends number | Date | string> {
+    x: X;
+    y: Y;
+}
+
+export function BarChart({
+    presentation,
+    series,
+    showLabels,
+    xAxisLabel,
+    yAxisLabel,
+    showLegend,
+    sortOrder,
+    style,
+    warningPrefix
+}: BarChartProps): ReactElement | null {
+    const [chartDimensions, setChartDimensions] = useState<{ height: number; width: number }>();
+
+    const warningMessagePrefix = useMemo(() => (warningPrefix ? warningPrefix + "i" : "I"), [warningPrefix]);
+
+    const dataTypesResult = useMemo(() => getDataTypes(series), [series]);
+
+    const sort = useMemo(() => ({ sortOrder, sortKey: "x" }), [sortOrder]);
+
+    const colorScale = { colorScale: "qualitative" } as { colorScale: "qualitative" };
+
+    const colorScaleProp = useMemo(
+        () => ({
+            ...colorScale
+        }),
+        [colorScale.colorScale]
+    );
+
+    const sortProp = useMemo(
+        () => ({
+            ...sort
+        }),
+        [sort]
+    );
+
+    const groupedOrStacked = useMemo(() => {
+        if (!dataTypesResult || dataTypesResult instanceof Error) {
+            return null;
+        }
+
+        // datum.y is actually the X axis data points due to the way Victory handles horizontal charts
+        const bars = series.map((series, index) => (
+            <VictoryBar
+                horizontal
+                key={index}
+                data={series.dataPoints}
+                {...sortProp}
+                {...(showLabels ? { labels: ({ datum }: { datum: any }) => datum.y } : undefined)}
+            />
+        ));
+
+        if (presentation === "grouped") {
+            return (
+                // todo: configure offset
+                <VictoryGroup {...colorScaleProp}>{bars}</VictoryGroup>
+            );
+        }
+
+        return <VictoryStack {...colorScaleProp}>{bars}</VictoryStack>;
+    }, [dataTypesResult, series, style, warningMessagePrefix]);
```

Comment:
sortProps, showLabels, presentation and colorScaleProp are missing in the deps

Reply:
I'll look into it

Reply:
Indeed, they should be added. Thanks for the catch!

Reply:
👍